generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -189,16 +203,21 @@ public OpaqueExpressionGenerator(IRandom generator, GenerationParams generationP
     if (!shadingLanguageVersion.supportedBitwiseOperations()) {
       return Optional.empty();
     }
-    final BinOp operator = generator.nextBoolean() ? BinOp.SHL : BinOp.SHR;
-    // The maximum safe value to shift zero without potentially causing undefined behavior is
-    // between zero and eight bits (as nine bits is the minimum precision of an integer in GLSL).
-    // If we're shifting one, we want to be able to shift backwards without losing information, and
-    // so our maximum shift value is seven bits.
-    final int bound = isZero ? 10 : 9;
-    final int maxValue = generator.nextInt(bound);
+    // The minimum precision for a lowp integer in GLSL is 9 bits (with one reserved for a sign
+    // bit if the integer is signed) - we don't have to worry about losing information if we're
+    // shifting zero, but shifting more than 8 bits may result in undefined behavior.
```

Comment:
The name "maxShiftForZero" isn't right; the max shift for zero is actually 8.  I would call the variable "minBitsForLowpInt".

Reply:
Done